### PR TITLE
Add tests for executeQuery BEGIN pipelining

### PR DESCRIPTION
**Pull request by robsdedude**

### CODE REVIEW

**Review by bigmontz — APPROVED** [review]
🔗

File: tests/stub/optimizations/test_optimizations.py [hunk 1]
```diff
@@ -55,6 +59,51 @@ def test():
...
+        driver = Driver(self._backend, f"bolt://{self._server.address}", auth,
+                        max_tx_retry_time_ms=1500)
+
+        with self.assertRaises(types.DriverError) as exc:
+            driver.execute_query("RETURN 1 AS n")
+
+        driver_name = get_driver_name()
+        if driver_name in ["python"]:
+            expected_error = "<class 'neo4j.exceptions.ServiceUnavailable'>"
+        else:
```
Comment (bigmontz) [inline tests/stub/optimizations/test_optimizations.py]:
```suggestion
        elif driver_name in ["javascript"]:
            expected_error = None
            self.assertEqual("ServiceUnavailable", exc.exception.code)
        else:
```

File: tests/stub/optimizations/test_optimizations.py [hunk 2]
```diff
@@ -55,6 +59,51 @@ def test():
...
+        driver.execute_query("RETURN 1 AS n")
+
+        driver.close()
+        self._server.done()
+
+    @driver_feature(types.Feature.API_DRIVER_EXECUTE_QUERY,
+                    types.Feature.OPT_EXECUTE_QUERY_PIPELINING,
+                    types.Feature.BOLT_5_3)
+    def test_execute_query_does_not_pipeline_commit(self):
+        script = "commit_pipeline.script"
```
Comment (thelonelyvulpes) [inline tests/stub/optimizations/test_optimizations.py]: This test seems wrong to me, we are verifying that bolt stub shuts down when sent a message in the wrong order.

Comment (thelonelyvulpes) [reply tests/stub/optimizations/test_optimizations.py:82]: this is a case that is implicitly tested, by the above test

Comment (thelonelyvulpes) [reply tests/stub/optimizations/test_optimizations.py]: this is the only reliable means of testing this behaviour(currently) so I withdraw this comment.

**Review by thelonelyvulpes — APPROVED** [review]

**Pull request merged**